### PR TITLE
[Feature][Engine UI] partial pages of the engine

### LINKED ISSUE #6932: [Feature][Zeta Engine UI] Add zeta engine UI.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

https://summer-ospp.ac.cn/org/prodetail/248940398?list=org&navpage=org

## Project Overview

**Project Name**: Zeta Web UI Page Management

## 1. Project Background

Currently, SeaTunnel's self-developed engine, Zeta, does not have a web UI that can intuitively display the system's health status and job execution status. Therefore, the main goal of this project is to develop a web UI for Zeta to address these issues.

### 2.1 Dashboard

**Description**: Provides an overview page that intuitively displays basic system information and job summaries.

**Features**:

- Displays the number of running jobs, completed jobs, and failed jobs.
- Shows system version and resource usage (number of slots, version, etc.).

### 2.2 Job Management

**Description**: Provides detailed job management features. Users can view, start, pause, and stop Zeta jobs.

**Features**:

- Lists all Zeta jobs, including job name, status, start time, duration, etc.
- Supports filtering jobs by status, name, time, etc.
- Job details page displaying detailed information about the job, including DAG (Directed Acyclic Graph), runtime, parallelism, etc.
- Supports operations to start, pause, and stop jobs.
- Provides job log viewing functionality with support for searching and filtering log content.

### 2.3 Task Management

**Description**: Provides management features for individual Zeta tasks, allowing users to view task execution status and performance metrics.

**Features**:

- Displays a list of tasks, including task name, status, start time, duration, etc.
- Task details page showing detailed information and execution status of the task.
- Provides performance metrics for tasks (throughput, runtime, etc.).

### 2.4 Cluster Management

**Description**: Manages and monitors the Zeta cluster, viewing resource usage and health status of the cluster.

**Features**:

- Displays a list of cluster nodes, including node name, status, role (worker/master), CPU and memory usage, etc.
- Node details page showing detailed information and running status of the node.

## 3. Acceptance Criteria

1. Zeta UI functional design document.
2. Zeta UI code PR, which needs to be finally merged into the main branch.
3. Zeta UI functionality meets the requirements and runs normally.
4. Zeta UI usage documentation.

## 4. Reference

https://seatunnel.apache.org/docs/2.3.5/seatunnel-engine/about

https://seatunnel.apache.org/docs/2.3.5/seatunnel-engine/rest-api

> 中文描述

## 项目概述

**项目名称**: SeaTunnel Zeta Engine UI 开发

## 1. 项目背景

目前SeaTunnel 自研引擎Zeta还没有一个 ui页面可以直观的展示系统的健康状况以及作业的执行情况，所以本题的主要目的就是为Zeta 开发一个web 页面解决这些问题。

### 2.1 仪表盘

**描述**: 提供一个总览页面，可以直观展示系统基本信息和作业的总览。

**功能点**:

- 显示正在运行的作业数、已完成的作业数、失败的作业数。
- 展示系统版本和资源使用情况（slot数、版本信息等）。

### 2.2 作业管理

**描述**: 提供详细的作业管理功能，用户可以查看、启动、暂停和停止Zeta作业。

**功能点**:

- 列表展示所有 Zeta 作业，包含作业名称、状态、开始时间、持续时间等信息。
- 支持按状态、名称、时间等条件筛选作业。
- 作业详情页面，展示作业的详细信息，包括 DAG（有向无环图）、运行时长、并行度等。
- 支持启动、暂停和停止作业的操作。
- 提供作业日志查看功能，支持搜索和过滤日志内容。

### 2.3 任务管理

**描述**: 提供对单个 Zeta 任务的管理功能，查看任务的执行情况和性能指标。

**功能点**:

- 展示任务列表，包括任务名称、状态、开始时间、持续时间等。
- 任务详情页面，展示任务的详细信息和执行情况。
- 提供任务的性能指标（吞吐量等信息）。

### 2.4 集群管理

**描述**: 监控 Zeta 集群，查看集群的资源使用情况和健康状态。

**功能点**:

- 展示集群的节点列表，包含节点名称、状态、角色（worker/master）、CPU 和内存使用情况等。
- 节点详情页面，展示节点的详细信息和运行状态。

## 3.完成标准

1.  Zeta UI功能设计文档 
2. Zeta UI代码PR, PR需要最终merge到主分支 
3. Zeta UI功能满足需求说明，正常运行 
4. Zeta UI的使用文档

## 4.参考

https://seatunnel.apache.org/docs/2.3.5/seatunnel-engine/about

https://seatunnel.apache.org/docs/2.3.5/seatunnel-engine/rest-api

https://github.com/apache/seatunnel/issues/6932

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by QiaoJ-Chen**
This project is used to display engine information and has now completed the overview page, job page, and manager page.

### CODE REVIEW

**Comment by QiaoJ-Chen** [comment]
once again

**Review by wuchunfu — COMMENTED** [review]
@QiaoJ-Chen These packaged dist files should not be submitted

**Comment by Hisoka-X** [comment]
Shall we need add test case to build ui when creating PR?

**Comment by liugddx** [comment]
> Shall we need add test case to build ui when creating PR?

Is there any good way to test UI?

**Comment by Hisoka-X** [comment]
> Is there any good way to test UI?

1. Make sure it can be build normally. Just like we did on seatunnel-website. https://github.com/apache/seatunnel/blob/dev/.github/workflows/documents.yml#L50
2. Add first test case with https://vitest.dev/guide/snapshot.html

File: seatunnel-engine/seatunnel-engine-ui/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Apache SeaTunnel Engine UI
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-ui/README.md]: Please update the doc. How to use it?

**Comment by dailai** [comment]
Can this feature integrate with this api #7818 to make searching logs easier.

**Comment by liugddx** [comment]
> Can this feature integrate with this api #7818 to make searching logs easier.

Agreed, we can integrate more capabilities

**Comment by Hisoka-X** [comment]
cc @baicie

File: seatunnel-engine/seatunnel-engine-ui/cypress/e2e/example.cy.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// https://on.cypress.io/api
+
+describe('My First Test', () => {
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-ui/cypress/e2e/example.cy.ts:20]: Hi, how to trigger test case in ci. I'm not familiar with the web.

Comment (QiaoJ-Chen) [reply seatunnel-engine/seatunnel-engine-ui/cypress/e2e/example.cy.ts:20]: Use "npm run test" to run the test cases. I did not use Cypress and have just removed it. Now, Vitest is being used as the testing scaffold.

File: seatunnel-engine/seatunnel-engine-ui/README.md (hunk 1 above)
Comment (QiaoJ-Chen) [reply seatunnel-engine/seatunnel-engine-ui/README.md]: The README.md has been updated

**Review by Hisoka-X — COMMENTED** [review]
Please add doc to introduce the new ui in https://github.com/apache/seatunnel/tree/c0f27c2f76c6562b18a7b4c76d2c7ef4452f4b4a/docs/en/seatunnel-engine

File: docs/en/seatunnel-engine/web-ui.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Apache SeaTunnel Web UI Documentation
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/web-ui.md:1]: Please update https://github.com/apache/seatunnel/blob/15431c94d9b9f34839173d67c1b638f058f58722/docs/sidebars.js#L209

File: docs/en/seatunnel-engine/web-ui.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Overview
+
+The Web UI of Apache SeaTunnel offers a user-friendly interface for monitoring and managing SeaTunnel jobs. Through the Web UI, users can view real-time information on currently running jobs, finished jobs, and the status of worker and master nodes within the cluster. The main functional modules include Jobs, Workers, and Master, each providing detailed status information and operational options to help users efficiently manage and optimize their data processing workflows.
+![workflow.png](../../images/ui/overview.png)
+
+## Jobs
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/web-ui.md:23]: Quick question: How can I access it?

File: docs/en/seatunnel-engine/web-ui.md (hunk 3 above)
Comment (QiaoJ-Chen) [reply docs/en/seatunnel-engine/web-ui.md:1]: Okay, it has been updated just now

File: docs/en/seatunnel-engine/web-ui.md (hunk 4 above)
Comment (QiaoJ-Chen) [reply docs/en/seatunnel-engine/web-ui.md:23]: Sorry, images was ignored, I've just resolved it

File: docs/en/seatunnel-engine/web-ui.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Apache SeaTunnel Web UI Documentation
+
+## Access
+
+Before accessing the web ui we need to enable the http rest api. first you need to configure it in the `seatunnel.yaml` configuration file
+``
+http.
+enable-http: true
+port: 8080
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/web-ui.md]:
```suggestion
seatunnel:
  engine:
    http:
      enable-http: true
      port: 8080
```

File: docs/en/seatunnel-engine/web-ui.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+```
+Then visit http://ip:8080/#/overview
+```
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/web-ui.md:14]:
```suggestion
```

File: docs/en/seatunnel-engine/web-ui.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# Apache SeaTunnel Web UI Documentation
+
+## Access
+
+Before accessing the web ui we need to enable the http rest api. first you need to configure it in the `seatunnel.yaml` configuration file
+``
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/web-ui.md]: change `` to ```

Comment (QiaoJ-Chen) [reply docs/en/seatunnel-engine/web-ui.md]: I just took care of it

File: docs/en/seatunnel-engine/web-ui.md (hunk 6 above)
Comment (QiaoJ-Chen) [reply docs/en/seatunnel-engine/web-ui.md:14]: Sorry，I just took care of it

File: docs/en/seatunnel-engine/web-ui.md (hunk 5 above)
Comment (QiaoJ-Chen) [reply docs/en/seatunnel-engine/web-ui.md]: it had been updated

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @QiaoJ-Chen and @liugddx!

**Review by liugddx — APPROVED** [review]

**Pull request merged**